File: 009.pl
#!/usr/bin/perl
use v5.12;
use warnings;

sub PPT_sum {
    my $target = shift;
    foreach my $x (grep { $_ % 2 } 1 .. sqrt $target) {
        foreach my $y (grep { !($_ % 2) } $x+1 .. sqrt $target) {
            my ($sa, $sb, $sc) = ($y**2 - $x**2, 2 * $x * $y, $x**2 + $y**2);
            return $sa * $sb * $sc if $sa + $sb + $sc == $target;
        }
    }
}

say PPT_sum(1000);

File: 009.py
#!/usr/bin/env python
#-*- coding:utf-8 -*-

target = 1000

limit = int(target**.5)
for i in xrange(1, limit, 2):
    for j in xrange(i + 1, limit, 2):
        a, b, c = j**2 - i**2, 2 * i * j, i**2 + j**2
        if a + b + c == target:
            print(a * b * c)
            break
